packages/client/src/internals/TRPCClient.ts

```diff
@@ -146,7 +139,8 @@ export class TRPCClient<TRouter extends AnyRouter> {
 
     const getLinks = (): OperationLink<TRouter>[] => {
       if ('links' in opts) {
-        return opts.links.map((link) => link(this.runtime));
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        return opts.links!.map((link) => link(this.runtime));
```

Comment:
The above triggers the need for this which isn't ideal.

Reply:
you can just do `if(opts.links)`

Reply:
yea I guess. Ideally it would infer that links will always be defined if it's passed. Maybe some typescript guru knows how?